### DISCUSSION TITLE
Set NodeTag on initial node

### DISCUSSION THREAD

**Discussion by rwkarg** [ravendb/ravendb#15181] [Q&A] [answered]
When adding new nodes to an existing cluster I can specify the node tag:
https://github.com/ravendb/ravendb/blob/b4add7c253da20a5bf1a630e1910501189d0a0f4/src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs#L284-L288

But I don't see a way to either set a node tag on bootstrap or change a node's tag later (I assume it can't be modified after joining the cluster).
Bootstrap seems to have a parameter that default the node tag to "A", but the API doesn't expose a way to override that:
https://github.com/ravendb/ravendb/blob/b4add7c253da20a5bf1a630e1910501189d0a0f4/src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs#L277-L282

The only way around this I can see is to bootstrap the cluster, add a second node, remove the first node, rebuild/clear data on first node, rejoin the first node to the cluster from the second node (specifying a node tag) which isn't a great workflow.

**Comment by ayende** [comment]
A PR for this would be great.

However, what is the actual scenario that you need this for? Why do you care what the tag on the node is?

**Reply by ayende** [reply]
Also, you can do that using the Complete Setup option, right now. But again, the question is why?

**Reply by rwkarg** [reply]
Convenience and consistency of node tag with the logical underlying host. Also, we have some databases that are localized to specific hosts in the cluster and having the location as part of the node tag makes that management easier. Not our exact case, but imagine a `DE` zone for german data/users with nodes `DEA` and `DEB` while there are `FRA`, `FRB`, `UKA`, `UKB` nodes for other data, but there are still some cluster-wide operations for other data that need to occur across all of those, so they're in the same cluster.

**Reply by rwkarg** [reply]
PR opened: https://github.com/ravendb/ravendb/pull/15207

**Comment by ppekrol** [comment, marked as answer]
PR https://github.com/ravendb/ravendb/pull/15207 got merged
